flashlight/fl/nn/modules/Module.h

```diff
@@ -177,6 +200,38 @@ class BinaryModule : public Module {
  private:
   FL_SAVE_LOAD_WITH_BASE(Module)
 };
+
+/**
+ * Casts the modulePtr to T* and calls the copy constructor to clone the module,
+ * returning a smart pointer.
+ *
+ * @param modulePtr The module pointer to clone
+ * @return The new cloned T module pointer
+ */
+template <typename T, template <typename, typename> class SmartPtr, typename D>
+std::unique_ptr<T> cloneModulePtr(const SmartPtr<Module, D>& modulePtr) {
```

Comment:
To summarize above discussion — let's just remove `cloneModulePtr` and `cloneModuleAs`. We can test `Module::clone()` is isolation, cast in tests, and go from there

Reply:
Sounds good. It will be easy to add later if desired.